src/puppetlabs/puppetdb/command.clj

```diff
@@ -296,25 +296,35 @@
 
 ;; Fact replacement
 
+(defn rm-facts-regex [facts-blacklist fact-map]
+  (let [blacklisted? (fn [fact-name]
+                       (some #(re-matches % fact-name) facts-blacklist))]
+    (apply dissoc fact-map (filter blacklisted? (keys fact-map)))))
+
 (defn replace-facts*
-  [{:keys [payload id received] :as command} start-time db facts-blacklist]
+  [{:keys [payload id received] :as command} start-time db
+   {:keys [facts-blacklist facts-blacklist-type] :as blacklist-config}]
   (let [{:keys [certname package_inventory] :as fact-data} payload
         producer-timestamp (:producer_timestamp fact-data)
+        rm-blacklist (when (seq blacklist-config)
+                       (case facts-blacklist-type
+                         "regex" (partial rm-facts-regex facts-blacklist)
+                         "literal" #(apply dissoc % facts-blacklist)))
```

Comment:
*If* blacklist-config will always be nil when there isn't one, then realized we could also just do this, also replacing the enclosing `when`: `(case facts-blacklist-type ... "regex"  ... "literal" ... nil identity)` and then below:
```
(cond-> (rm-blacklisted fact-data)
  (seq package_inventory) (update :package_inventory distinct))
```